Mathlib/Dynamics/TopologicalEntropy/DynamicalEntourage.lean

```diff
@@ -0,0 +1,132 @@
+/-
+Copyright (c) 2024 Damien Thomine. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damien Thomine, Pietro Monticone
+-/
+import Mathlib.Order.Interval.Finset.Nat
+import Mathlib.Order.OmegaCompletePartialOrder
+import Mathlib.Topology.UniformSpace.Basic
+
+/-!
+# Dynamical entourages
+Bowen-Dinaburg's definition of topological entropy of a transformation `T` in a metric space
+`(X, d)` relies on the so-called dynamical balls. These balls are sets
+`B (x, ε, n) = { y | ∀ k < n, d(T^[k] x, T^[k] y) < ε }`.
+
+We implement Bowen-Dinaburg's definitions in the more general context of uniform spaces. Dynamical
+balls are replaced by what we call dynamical entourages. This file collects all general lemmas
+about these objects.
+
+## Main definitions
+- `dynEntourage`: dynamical entourage associated with a given transformation `T`, uniformity `U`
+and time `n`.
+
+## Tags
+entropy
+
+## TODO
+Once #PR14718 has passed, add product of uniformities.
+
+In the context of (pseudo-e)metric spaces, relate the usual definition of dynamical balls with
+these dynamical uniformities.
+-/
+
+namespace DynamicalEntourage
```

Comment:
```suggestion
namespace Dynamics
```
You don't need to have a very specific namespace, you just want to make sure that you're not spoiling the root namespace.